src/psij/executors/batch/slurm/slurm.mustache

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+{{#job.name}}
+#SBATCH --job-name="{{.}}"
+{{/job.name}}
+
+{{#job.spec.directory}}
+#SBATCH --chdir="{{.}}"
+{{/job.spec.directory}}
+
+{{#job.spec.inherit_environment}}
+#SBATCH --export=ALL
+{{/job.spec.inherit_environment}}
+{{^job.spec.inherit_environment}}
+#SBATCH --export=NONE
+{{/job.spec.inherit_environment}}
+
+{{#job.spec.resources}}
+    {{#exclusive_node_use}}
+#SBATCH --exclusive
+    {{/exclusive_node_use}}
+
+    {{#node_count}}
+#SBATCH --nodes={{.}}
+    {{/node_count}}
+
+    {{#process_count}}
+#SBATCH --ntasks={{.}}
+    {{/process_count}}
+
+    {{#processes_per_node}}
+#SBATCH --ntasks-per-node={{.}}
+    {{/processes_per_node}}
+
+    {{#gpu_cores_per_process}}
+#SBATCH --gpus_per_task={{.}}
+    {{/gpu_cores_per_process}}
+
+    {{#cpu_cores_per_process}}
+        {{!SLURM has different meanings for 'CPU', depending on the exact allocation plugin being
+        used. The default plugin uses the literal meaning for 'CPU' - physical processor. The
+        consumable resource allocation plugin, on the other hand, equates 'CPU' to a thread (if
+        hyperthreaded CPUs are used) or CPU core (for non-hyperthreaded CPUs).}}
+#SBATCH --cpus_per_task={{.}}
+    {{/cpu_cores_per_process}}
+{{/job.spec.resources}}
+
+{{#job.spec.attributes}}
+    {{#duration}}
+#SLURM --time={{#psij.lib.walltime_to_minutes}}{{duration}}{{/psij.lib.walltime_to_minutes}}
+    {{/duration}}
+
+    {{#queue_name}}
+#SLURM --partition="{{.}}"
+    {{/queue_name}}
+
+    {{#project_name}}
+#SLURM --account="{{.}}"
+    {{/project_name}}
+
+    {{#reservation_id}}
+#SLURM --reservation="{{.}}"
+    {{/reservation_id}}
+
+    {{#custom_attributes.slurm}}
+#SBATCH --{{key}}="{{value}}"
+    {{/custom_attributes.slurm}}
+
+{{/job.spec.attributes}}
+
+{{!redirect output here instead of through #SBATCH directive since SLURM_JOB_ID is not available
+when the directives are evaluated; the reason for using the job id in the first place being the
+same as for the exit code file.}}
```

Comment:
We though could consider using the jpsi job IDs for file identifiers - we assume uniqueness there as well, and it would keep the name consistent with `submit_file_path` from [here](https://github.com/ExaWorks/psi-j-python/pull/49/files#diff-c6b591d8329e3042794fe7dda7c419fc4572f488c6665f3383e0d2688019ae98R142).

Reply:
That was my first instinct. But that wouldn't be robust with attach() since the attached job id would generally be different from the initial job id whereas the native id would not.

Reply:
I'll keep as is for reasons outlined above.